code/game/turfs/open/_open.dm

```diff
@@ -322,12 +318,18 @@
 		LAZYINITLIST(damage_overlays)
 	if(burnt && !force || use_burnt_literal)
 		if(use_burnt_literal)
-			icon_state = pick(burnt_states)
+			icon_state = pick(get_burnt_states())
 		return
-	if(length(burnt_states))
-		var/burnt_state = pick(burnt_states)
+	if(length(get_burnt_states()))
+		var/burnt_state = pick(get_burnt_states())
 		//Add some burnt shit
 		var/icon/burnt_overlay = icon(broken_icon, "burnt_[burnt_state]")
 		add_filter("brunt_overlay", 4, layering_filter(icon = burnt_overlay))
 		damage_overlays += "brunt_overlay"
 	burnt = TRUE
+
+/turf/open/proc/get_damage_states()
+	return GLOB.default_turf_damage
```

Comment:
not sure that using procs is the best idea here. They kinda perform slow. I'd recommend some type of #define abomination that uses a var on the area/turf.

Reply:
It's what TG did. Mothblocks did the same solution but returns every new list(), not glob.

Reply:
I don't wanna. This is the way